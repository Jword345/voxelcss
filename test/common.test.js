'use strict';

const {
    multiplyMatrices,
    generateRotationMatrix
} = require('../lib/common');
const {PI} = Math;

describe('Common utilities', function() {
    it('can perform matrix multiplication (3x3)', function() {
        let x = [1, 0, 0];
        let y = [0, 1, 0];
        let z = [0, 0, 1];
        let a = [x, y, z];
        let b = [z, y, x];
        let c = [z, x, y];
        expect(multiplyMatrices(a, b)).toMatchSnapshot();
        expect(multiplyMatrices(a, c)).toMatchSnapshot();
        expect(multiplyMatrices(b, c)).toMatchSnapshot();
    });
    it('can generate rotation matrices', function() {
        expect(generateRotationMatrix(0, 0, 0)).toMatchSnapshot();
        expect(generateRotationMatrix(PI, PI, PI)).toMatchSnapshot();
        expect(generateRotationMatrix(PI, 0, 0)).toMatchSnapshot();
        expect(generateRotationMatrix(0, PI, 0)).toMatchSnapshot();
        expect(generateRotationMatrix(0, 0, PI)).toMatchSnapshot();
        expect(generateRotationMatrix(-PI, -PI, -PI)).toMatchSnapshot();
    });

});
