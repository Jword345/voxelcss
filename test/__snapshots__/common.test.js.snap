// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Common utilities can generate rotation matrices 1`] = `
Array [
  Array [
    1,
    0,
    0,
  ],
  Array [
    0,
    1,
    0,
  ],
  Array [
    0,
    0,
    1,
  ],
]
`;

exports[`Common utilities can generate rotation matrices 2`] = `
Array [
  Array [
    1,
    1.224646799147353e-16,
    1.2246467991473535e-16,
  ],
  Array [
    -1.2246467991473532e-16,
    1,
    1.224646799147353e-16,
  ],
  Array [
    -1.2246467991473532e-16,
    -1.2246467991473532e-16,
    1,
  ],
]
`;

exports[`Common utilities can generate rotation matrices 3`] = `
Array [
  Array [
    1,
    0,
    0,
  ],
  Array [
    0,
    -1,
    -1.2246467991473532e-16,
  ],
  Array [
    0,
    1.2246467991473532e-16,
    -1,
  ],
]
`;

exports[`Common utilities can generate rotation matrices 4`] = `
Array [
  Array [
    -1,
    0,
    1.2246467991473532e-16,
  ],
  Array [
    0,
    1,
    0,
  ],
  Array [
    -1.2246467991473532e-16,
    0,
    -1,
  ],
]
`;

exports[`Common utilities can generate rotation matrices 5`] = `
Array [
  Array [
    -1,
    -1.2246467991473532e-16,
    0,
  ],
  Array [
    1.2246467991473532e-16,
    -1,
    0,
  ],
  Array [
    0,
    0,
    1,
  ],
]
`;

exports[`Common utilities can generate rotation matrices 6`] = `
Array [
  Array [
    1,
    -1.2246467991473535e-16,
    -1.224646799147353e-16,
  ],
  Array [
    1.2246467991473532e-16,
    1,
    -1.2246467991473535e-16,
  ],
  Array [
    1.2246467991473532e-16,
    1.2246467991473532e-16,
    1,
  ],
]
`;

exports[`Common utilities can perform matrix multiplication (3x3) 1`] = `
Array [
  Array [
    0,
    0,
    1,
  ],
  Array [
    0,
    1,
    0,
  ],
  Array [
    1,
    0,
    0,
  ],
]
`;

exports[`Common utilities can perform matrix multiplication (3x3) 2`] = `
Array [
  Array [
    0,
    0,
    1,
  ],
  Array [
    1,
    0,
    0,
  ],
  Array [
    0,
    1,
    0,
  ],
]
`;

exports[`Common utilities can perform matrix multiplication (3x3) 3`] = `
Array [
  Array [
    0,
    1,
    0,
  ],
  Array [
    1,
    0,
    0,
  ],
  Array [
    0,
    0,
    1,
  ],
]
`;
